### DISCUSSION TITLE
VNA-X, measure IMD3 with the LibreVNA

### DISCUSSION THREAD

**Discussion by satfan52** [jankae/LibreVNA#227] [Ideas]
Professional VNAs of the VectorStar line of Anritsu can be used to characterize non-linear devices such as power amplifiers. In addition to measuring S parameters, they can also generate a two tone signal centered on a given frequency and of a given signal power;  then measure the corresponding IMD3. This feature is unavailable for hobbiest so far on low cost equipment, adding it to the LibreVNA would make it the first true VNA-X for hobbiest. LibreVNA can already do power sweeps and be used as a basic spectrum analyzer (if my understanding is right), so adding the IMD3 measurement would be a great asset and the last one missing to offer basic VNA-X capabilities to hobbiest.

**Comment by jankae** [comment]
This is not possible with the LibreVNA. As you mentioned it requires the simultaneous generation of two signals and the LibreVNA only has one source. You would either need a second source or a way to FM modulate the existing source. Both options require additional hardware and are not very cheap. I don't really want to include that by default in any new hardware versions. Maybe I can come up with some extension module to cover this functionality.

**Reply by satfan52** [reply]
If only one source is available and if it is not possible to modulate a DSB signal, maybe an approach with two LibreVNA used in T|Rmode using  a combiner on port 1 to generate the two tone  and a spliter at the ouput of the DUT to send the test signal  to port 2 of each Libre VNA would work ? 

If I as suspect it only one VNA  receiver is needed to measure the IMD,  maybe the second receiver of the second VNA can be connected to the input combiner 2 too, to measure the input signal and make adequate corrections in the calculation of IMD  (if needed)

The PC app would have to be modified to be able connect to both LibreVNAs and to calculate the IMD products.

I know it may seem an overeach to use two librevnas but on the other hand, if IMD measurements can be automated it is worth the investment of a second LibreVNA  in my view (considering the LibreVNA is not so expensive) 

Finally, for optimal return on investment, the second librevna could also be used to create a virtual 4 port VNA provided the PC app  is modified adequately too.

**Comment by jankae** [comment]
> maybe an approach with two LibreVNA used in T|Rmode using a combiner on port 1 to generate the two tone and a spliter at the ouput of the DUT to send the test signal to port 2 of each Libre VNA would work ?

Yes, from a hardware perspective this should work. The software is almost there: you can measure IMD3 in the spectrum analyzer mode already but you need to provide all sources externally. You have a tracking generator in the spectrum analyzer mode as well but it always tracks the measured frequency (with an optional offset). For this measurement to work, you would need a "non-tracking" generator, that is a fixed frequency.
Neither the source nor the receiver levels in the LibreVNA are particular accurate, so I am somewhat doubting how useful this measurement would turn out (unless you calibrate them).

> Finally, for optimal return on investment, the second librevna could also be used to create a virtual 4 port VNA provided the PC app is modified adequately too.

That is already possible, you just lose phase information for transmission measurements between two physical VNAs (e.g. phase information is there for S21 but not for S31 if port 1/2 are on one VNA and port 3/4 on the other). This is another hardware limitation as the LO can't be shared between the VNAs.

**Comment by satfan52** [comment]
Thanks.

I understand very well that a single LibreVNA could be used in spectrum
analyzer mode with two external power sources  to measure the IMD3 in dBc.
This is what  I do it with  the spectrum analyzer that  I own
already....but this is a slow and extremely  painful procedure when it has
to be repeated at different frequencies and different power levels. This
very painfull issue is very well known by all people who characterize
amplifiers and can't afford professional VNA-X equipment costing thousands
of Euros.

Doing such measurements with rhe LibreVNA in spectrum analyzer mode would
of course require a proper linearity of the LibreVNA spectrum analyzer
itself which I assume is the case because   otherwise,  the IMD3 measured
will  be the one of the LibreVNA instead of the DUT one ! If the linearity
is not good enough,  there is indeed no point using the LibreVna in
spectrum analyzer mode for such IMD measurements.

My initial point was about automating the IMD3 measurement over a given
frequency sweep (at a given power) or over a power sweep (at a given
frequency). This is where I find  tremendous  value in  investing  in two
LibreVNAs provided the software allows to automate such IMD3 measurents
'(in dBc) over frequency or  powersweeps.

Thank you for you patience and your great  work on the LibreVNA

Regards
Peter


Le mer. 15 nov. 2023, 13:34, Jan Käberich ***@***.***> a
écrit :
